department-of-veterans-affairs/caseflow
# Pamatya/appeals 23410

Resolves APPEALS-23410 (https://jira.devops.va.gov/browse/JIRA-23410)

# Description
1.  move few methods from appeals_controller to Application controller so that both decision review controller can use it. 
2.  added a method and route called 'update_power_of_attorney' in decision_reviews so that POA information can be updated.
3.  removed update_POA link in disposition.jsx coz its not needed for vha. 
4. rspec and serializer fix 

## Acceptance Criteria
- [ ] Refresh POA button should update the POA information
- [ ] POA Alert should be displayed with different message.
- [ ] Last refreshed message should be updated.

## Testing Plan

1. Go to [Jira Issue/Test Plan Link](https://jira.devops.va.gov/browse/JIRA-23410).
2.  Login User : Any user which has access to VHA decision review. you can use - ACBAUERVVHA or create user
````
vha_user = FactoryBot.create(:user)
vha_org = BusinessLine.find_by_url( "vha")
vha_org.add_user(vha_user)
````
3. Create Higher level Review and Supplemental review
````
veteran = FactoryBot.create(:veteran) 
FactoryBot.create(:higher_level_review,
             :with_vha_issue,
             :with_end_product_establishment,
             :create_business_line,
             benefit_type: "vha",
             veteran: veteran,
             number_of_claimants: 1)

sc = FactoryBot.create(:supplemental_claim, :processed, :with_vha_issue, :with_end_product_establishment, benefit_type: "vha", number_of_claimants: 1)
sc.create_business_line_tasks!
````
4. redirect to decision_reviews/vha after logging in.
5. on the table, click on the claimant name to view the decision review detail .
6. Check if Appellant's Power Of attorney section is present. under veteran description and above decision section.
7. Verify if the claimant detail match and Verify if Refresh POA button is present in all cases except unrecognized representative.
8.  Clicking on the refresh POA button should give Alerts like -> Information is current at this time. Please try again in 10 minutes.
9. 


# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---
![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/f7b3d5c8-269e-444a-90c9-fec95d0445a5)|![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/d442033a-76f6-4728-9479-0417ec29b545)![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/03300482-dda7-4e09-a6a1-32d680315051)![image](https://github.com/department-of-veterans-affairs/caseflow/assets/122557351/35cc4256-9927-4d13-bc3d-1f3c485182ac)



## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
